### ISSUE TITLE
Worker queue system is broken

### ISSUE THREAD

**Issue by thomasmorgan** [Dallinger/Dallinger#270]
The worker dyno fails to start properly and crashes. This causes errors when notifications arrive and the web dyno tries to put them on the queue. May be relevant to #269 

## Expected Behavior
The worker dyno should not crash and so it should be possible to place items on the queue

## Current Behavior
The worker crashes when it tries to setup:
```
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:  Traceback (most recent call last): 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "worker.py", line 22, in <module> 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      worker.work() 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/rq/worker.py", line 423, in work 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      self.register_birth() 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/rq/worker.py", line 242, in register_birth 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      if self.connection.exists(self.key) and \ 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 855, in exists 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      return self.execute_command('EXISTS', name) 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 578, in execute_command 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      connection.send_command(*args) 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 563, in send_command 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      self.send_packed_command(self.pack_command(*args)) 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 538, in send_packed_command 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      self.connect() 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 442, in connect 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:      raise ConnectionError(self._error_message(e)) 
Oct 19 13:44:08 dlgr-471fc00c app/worker.2:  redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused. 
```
This causes errors when items are put on the queue:
```
Oct 19 13:46:40 dlgr-471fc00c app/web.1:  2016-10-19 20:46:40,459 Exception on /notifications [GET] 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:  Traceback (most recent call last): 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      response = self.full_dispatch_request() 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      rv = self.handle_user_exception(e) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      reraise(exc_type, exc_value, tb) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      rv = self.dispatch_request() 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      return self.view_functions[rule.endpoint](**req.view_args) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/custom.py", line 1115, in api_notifications 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      q.enqueue(worker_function, event_type, assignment_id, None) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/rq/queue.py", line 263, in enqueue 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      job_id=job_id, at_front=at_front, meta=meta) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/rq/queue.py", line 216, in enqueue_call 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      job = self.enqueue_job(job, at_front=at_front) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/rq/queue.py", line 284, in enqueue_job 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      pipe.execute() 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 2641, in execute 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      return execute(conn, stack, raise_on_error) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/client.py", line 2495, in _execute_transaction 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      connection.send_packed_command(all_cmds) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 538, in send_packed_command 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      self.connect() 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:    File "/app/.heroku/python/lib/python2.7/site-packages/redis/connection.py", line 442, in connect 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:      raise ConnectionError(self._error_message(e)) 
Oct 19 13:46:40 dlgr-471fc00c app/web.1:  ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
```

In some instances the worker does eventually manage to start, but this is normally only after several minutes.

## Steps to Reproduce (for bugs)
Sandbox an experiment, play it through to send a submitted notification.

**Comment by thomasmorgan** [comment]
@suchow This is now fixed by having dallinger wait for redis to load, close?

**Comment by suchow** [comment]
Closed by #271.
